controllers/posts_controller.js

```diff
@@ -77,6 +77,19 @@ module.exports.search = async (req, res) => {
   }
 }
 
+module.exports.likePost = async ({params: {id = null}, user = null}, res) => {
+  try {
+    const postID = Number(id)
+    await postModel.likePost(postID, user.id)
+    res.status(200).json(new Response(null))
+  } catch (e) {
+    console.error(e)
+    res.status(500).json(new ErrorResponse(
+      [new Error('Failed to like this post')]
+    ))
```

Comment:
How did we make it to JavaLand with new ErrorResponse(new Error) ?

Reply:
See PR#23

Reply:
I see. Objects without methods is bad practice/overkill (confirm @koosoku). I'd suggest to make your Error and ErrorResponse stuff into helpers:

```js
// res is injected
Response.OK(res, foo); // does res.status(200).json({data: foo});
Response.NotFound(res, foo); // does res.status(404).json({data: null, error: foo});
```

Reply:
That's a good idea

Reply:
@jonathanGB What should I do for the Error class? Because I still want to make sure I'm not typing `{title: error}` every time, which is why there was an Error class in the first place.